File: dagger/src/main/java/com/cj/dagger/MainActivity.java
package com.cj.dagger;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import javax.inject.Inject;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;

public class MainActivity extends AppCompatActivity {
    @Inject
    public User user;
    @Inject
    public Retrofit retrofit;
//
//    @Inject
//    public ApiService apiService;
    @Inject
    OkHttpClient okHttpClient;
    @Inject
    OkHttpClient okHttpClient1;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        DaggerApplicationComponent.create().inject(this);
        Log.d("junchen", "onCreate: user--"+user);
        Log.d("junchen", "onCreate: okHttpClient--"+okHttpClient);
        Log.d("junchen", "onCreate: okHttpClient1--"+okHttpClient1);
        Log.d("junchen", "onCreate: retrofit--"+retrofit);
//        Log.d("junchen", "onCreate: apiService--"+apiService);
        startActivity(new Intent(this,SecondActivity.class));
    }
}

File: app/src/main/java/com/cj/kotlinjetpackapplication/MainActivity.kt
package com.cj.kotlinjetpackapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.cj.kotlinjetpackapplication.databinding.ActivityMainBinding
import dagger.hilt.android.AndroidEntryPoint
import retrofit2.Retrofit
import javax.inject.Inject

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    @Inject
    lateinit var truck:Truck
    @Inject
    lateinit var retrofit: Retrofit
    @Inject
    lateinit var person: Person

    @Inject
    lateinit var man: Man

    @Inject
    lateinit var women: Women

    @Inject
    lateinit var animal: Animal

    @Inject
    lateinit var goods: Goods
    val viewModel:MyViewModel by lazy { ViewModelProvider(this).get(MyViewModel::class.java) }
    private val binding by lazy {ActivityMainBinding.inflate(layoutInflater)}
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
        truck.deliver()
    }

    fun click(view: View) {
        binding.tv.text = "hahaha"
    }
}